packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/DatasetResource.java

```diff
@@ -99,7 +99,7 @@ public Response updateDataset(
 		@PathParam("id") final String id,
 		final Dataset dataset
 	) {
-		return datasetProxy.updateDataset(id, convertObjectToSnakeCaseJsonNode(dataset));
+		return datasetProxy.updateDataset(id, dataset);
```

Comment:
From previous comment, this is still needed to address casing issues.

Reply:
When I revert all the Java back I get: `"PATCH /datasets/50a218b4-b2e4-49bc-bd8a-d52e06835ac5 HTTP/1.1" 422 Unprocessable Entity` if I try to rename a dataset though. Not sure why this is the case when dataset upload works and that calls `updateDataset`

Reply:
Not sure if this makes a difference but I printed what the json looks like that's being passed to `updateDataset`

Here is it for uploading a csv:
`{"id":"57efd4ec-df85-4840-97fe-4683e70f51bf","columns":[{"name":"","annotations":[]},{"name":"date","annotations":[]},{"name":"I","annotations":[]},{"name":"R","annotations":[]},{"name":"D","annotations":[]},{"name":"V","annotations":[]},{"name":"H","annotations":[]},{"name":"I_0-9","annotations":[]},{"name":"I_10-19","annotations":[]},{"name":"I_20-29","annotations":[]},{"name":"I_30-39","annotations":[]},{"name":"I_40-49","annotations":[]},{"name":"I_50-59","annotations":[]},{"name":"I_60-69","annotations":[]},{"name":"I_70-79","annotations":[]},{"name":"N_0-9","annotations":[]},{"name":"N_10-19","annotations":[]},{"name":"N_20-29","annotations":[]},{"name":"N_30-39","annotations":[]},{"name":"N_40-49","annotations":[]},{"name":"N_50-59","annotations":[]},{"name":"N_60-69","annotations":[]},{"name":"N_70-79","annotations":[]},{"name":"N_80-89","annotations":[]},{"name":"N_90-99","annotations":[]},{"name":"N_100+","annotations":[]}]}`

Here it looks like this when I rename it:
`
{"id":"57efd4ec-df85-4840-97fe-4683e70f51bf","timestamp":"2023-08-28T17:42:25","username":"Emperor Adam","name":"usa-IRDVHN_age2","description":"usa-IRDVHN_age.csv","file_names":["usa-IRDVHN_age.csv"],"columns":[{"name":"","data_type":"UNKNOWN","annotations":[]},{"name":"date","data_type":"UNKNOWN","annotations":[]},{"name":"I","data_type":"UNKNOWN","annotations":[]},{"name":"R","data_type":"UNKNOWN","annotations":[]},{"name":"D","data_type":"UNKNOWN","annotations":[]},{"name":"V","data_type":"UNKNOWN","annotations":[]},{"name":"H","data_type":"UNKNOWN","annotations":[]},{"name":"I_0-9","data_type":"UNKNOWN","annotations":[]},{"name":"I_10-19","data_type":"UNKNOWN","annotations":[]},{"name":"I_20-29","data_type":"UNKNOWN","annotations":[]},{"name":"I_30-39","data_type":"UNKNOWN","annotations":[]},{"name":"I_40-49","data_type":"UNKNOWN","annotations":[]},{"name":"I_50-59","data_type":"UNKNOWN","annotations":[]},{"name":"I_60-69","data_type":"UNKNOWN","annotations":[]},{"name":"I_70-79","data_type":"UNKNOWN","annotations":[]},{"name":"N_0-9","data_type":"UNKNOWN","annotations":[]},{"name":"N_10-19","data_type":"UNKNOWN","annotations":[]},{"name":"N_20-29","data_type":"UNKNOWN","annotations":[]},{"name":"N_30-39","data_type":"UNKNOWN","annotations":[]},{"name":"N_40-49","data_type":"UNKNOWN","annotations":[]},{"name":"N_50-59","data_type":"UNKNOWN","annotations":[]},{"name":"N_60-69","data_type":"UNKNOWN","annotations":[]},{"name":"N_70-79","data_type":"UNKNOWN","annotations":[]},{"name":"N_80-89","data_type":"UNKNOWN","annotations":[]},{"name":"N_90-99","data_type":"UNKNOWN","annotations":[]},{"name":"N_100+","data_type":"UNKNOWN","annotations":[]}]}
`

Here is the reverted version where I am printing stuff https://github.com/DARPA-ASKEM/Terarium/commit/2a012c6dcd60d08d42723af8fbffab7a1fbc0321

Reply:
That looks right - if you don't pass it through the converter then you'd end up with `fileNames` instead of `file_names` for example.  I'm wondering if its the presence of the field `timestamp` that is messing things up since that should be a field only set by TDS. if you dont have that present does it work?

Reply:
it works when I remove `columns`!